### PR TITLE
OSDOCS-4686: Created a table for the VPC requirements

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
Enterprise-4.13+

Issue:
[OSDOCS-4686](https://issues.redhat.com/browse/OSDOCS-4686)

Link to docs preview:
- **[Creating a ROSA cluster with customizations](https://53699--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html#osd-aws-vpc-required-resources_rosa-sts-creating-a-cluster-with-customizations)**
- **[Creating a ROSA cluster quickly](https://53699--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.html#osd-aws-vpc-required-resources_rosa-sts-creating-a-cluster-quickly)**

QE review:
- [ ] QE has approved this change.

Additional information:
Created a new section for VPC requirements.

### CODE REVIEW

**Comment by bergerhoffer** [comment]
The `branch/enterprise-4.13` label has been added to this PR.

This is because your PR targets the `main` branch and is labeled for enterprise-4.12. And any PR going into `main` must also target the latest version branch (enterprise-4.13).

If the update in your PR does NOT apply to version 4.13 onward, please re-target this PR to go directly into the appropriate version branch or branches (enterprise-4.x) instead of main.

**Comment by codymant** [comment]
@EricPonvelle Here's a summary of my feedback on this task based on our conversation over Slack: 

- Assuming that default VPC configuration works for the new ROSA HyperShift topology, I recommend replacing the table in the section "Requirements for using non-PrivateLink AWS clusters" with a link to the Amazon VPC docs on default VPC configuration. This configuration has been validated to work with non-PrivateLink clusters for existing implementations: 

https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html
https://docs.aws.amazon.com/vpc/latest/userguide/working-with-vpcs.html

- Because the page "Creating an AWS PrivateLink cluster on ROSA" is focused on info needs for ROSA PrivateLink clusters, I believe the non-PrivateLink information is out of scope here. I recommend moving this section to the pages "Creating a ROSA cluster with STS using the default options" and "Creating a ROSA cluster with STS using customizations". STS is the recommended method for creating ROSA non-PrivateLink clusters. These pages guide users though creating public ROSA clusters, and VPC requirement information would be in scope for this type of content. 

https://53699--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.html

https://53699--docspreview.netlify.app/openshift-rosa/latest/rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.html

**Review by wgordon17 — APPROVED** [review]
LGTM

File: rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc [hunk 1]
```diff
@@ -30,6 +30,14 @@ include::modules/rosa-sts-understanding-aws-account-association.adoc[leveloffset
 
 * For detailed steps to create and link the {cluster-manager} and user IAM roles, see xref:../rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc#rosa-sts-associating-your-aws-account_rosa-sts-creating-a-cluster-quickly[Associating your AWS account with your Red Hat organization].
 
+include::modules/osd-aws-vpc-required-resources.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* For information about the default components required for an AWS cluster, see link:https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html[Default VPCs] in the AWS documentation.
```
Comment (mjpytlak) [inline rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc:38]:
```suggestion
* For more information about the default components required for an AWS cluster, see link:https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html[Default VPCs] in the AWS documentation.
```

File: rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.adoc [hunk 2]
```diff
@@ -30,6 +30,14 @@ include::modules/rosa-sts-arn-path-customization-for-iam-roles-and-policies.adoc
 //* TBC - ROSA CLI reference.
 
 include::modules/rosa-sts-support-considerations.adoc[leveloffset=+1]
+include::modules/osd-aws-vpc-required-resources.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* For information about the default components required for an AWS cluster, see link:https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html[Default VPCs] in the AWS documentation.
```
Comment (mjpytlak) [inline rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-with-customizations.adoc:38]:
```suggestion
* For more information about the default components required for an AWS cluster, see link:https://docs.aws.amazon.com/vpc/latest/userguide/default-vpc.html[Default VPCs] in the AWS documentation.
```

**Review by mjpytlak — COMMENTED** [review]
A few items for your consideration @EricPonvelle

**Pull request merged**